### ISSUE TITLE
Cannot fetch fields of @Group submodes.

### ISSUE THREAD

**Issue by Jean-Daniel** [vapor/fluent-kit#618]
### Describe the issue

Trying to use a field from a @Group struct in methods like `.field()`, `.all()`, etc fails with the error like `Instance method 'all' requires the types 'User' and 'GroupPropertyPath<User, FieldProperty<Pet, String>>.Model' (aka 'Pet') be equivalent.`

### Vapor version

1.49.0

### Operating system and version

macOS 14.7

### Swift version

Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)

### Steps to reproduce

Create a Vapor project.

Define the following model:

```swift 
final class Pet: Fields {
    // The pet's name.
    @Field(key: "name")
    var name: String

    // The type of pet.
    @Field(key: "type")
    var type: String

    // Creates a new, empty Pet.
    init() { }
}
final class User: Model {

  static var schema: String = "users"

  @ID(key: .id)
  var id: UUID?

  // The user's nested pet.
  @Group(key: "pet")
  var pet: Pet
}
```

Try to build a query using that model

```swift
func query(req: Request) {
  User.query(on: req.db).filter(\.$pet.$name == "Zizek").all(\.$pet.$name)
}
```

Build the project.

### Outcome

The build fails with the error:

```
Instance method 'all' requires the types 'User' and 'GroupPropertyPath<User, FieldProperty<Pet, String>>.Model' (aka 'Pet') be equivalent.
```

**Comment by 0xTim** [comment]
I don't think this is supported behaviour. The `@Group` is supposed to be all fetched so that we can populate it correctly. I'd say if you need to retrieve specific fields you should look at flattening the model

**Comment by gwynne** [comment]
Tim is correct; the `all()` method does not support deep keypaths - and while fixing it to do so would be simple, it would also require a source-breaking change and thus can't be done for Fluent 4.
